kge/util/dump.py

```diff
@@ -113,90 +113,174 @@ def _add_dump_trace_parser(subparsers_dump):
+            "A path to a file which contains lines in the format"
+            " 'new_key_name'='key_name'. For every line in the keys file, the command"
+            " will look for the value of 'key_name' in the trace entries and config and"
+            " add a respective column in the csv file with name 'new_key_name'."
+            " Additionally, for 'key_name' the special keys '$folder', '$machine'"
+            " '$checkpoint' and '$base_model' can be used. "
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--keys",
+        "-k",
+        nargs="*",
+        type=str,
+        help=(
+            "A list of 'key' entries (separated by space). Each 'key' has form"
+            " 'new_key_name=key_name' or 'key_name'. This will add columns as in the"
+            " --keysfile option. When only 'key_name' is provided, it is also used as"
+            " column name."
+        ),
+    )
     parser_dump_trace.add_argument(
         "--checkpoint",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If source is a path to a job folder and --checkpoint is set the best "
-            "(if present) or last checkpoint will be used to determine the job_id"
+            "If source is a path to a job folder and --checkpoint is set, the best"
+            " (if present) or last checkpoint will be used to determine the job_id from"
+            " where the tracefile will be processed backwards."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--job_id",
         default=False,
         help=(
-            "Specifies the training job id in the trace "
-            "from where to start processing backward"
+            "Specifies the training job id in the trace"
+            " from where to start processing the tracefile backward."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--max_epoch",
         default=False,
         help=(
-            "Specifies the epoch in the trace"
-            "from where to start processing backwards"
+            "Specifies the max epoch number in the tracefile"
+            " from where to start processing backwards. Can not be used with --truncate"
+            " when a checkpoint is provided."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--truncate",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If a checkpoint is used (by providing one explicitly as source or by "
-            "using --checkpoint), --truncate will define the max_epoch to process as"
-            "specified by the checkpoint"
+            "When a checkpoint is used (by providing one explicitly as source or by"
+            " using --checkpoint), --truncate will define the max_epoch number to"
+            " process as provided by the checkpoint. Can only be used when a checkpoint"
+            " is provided and it cannot be used together with --max_epoch."
```

Comment:
Likewise.

Reply:
Likewise. We discussed we want as default when a checkpoint is provided that sill all epochs of the job from that checkpoint would be included. But to also have to flexibility to only process up until the epoch which is provided from the checkpoint which is what --truncate does.

Reply:
And this one just --truncate

Reply:
It is called just --truncate already. What do you mean?

Reply:
Right.

:)